backend/YouTubeV2/Controllers/DonationController.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System.ComponentModel.DataAnnotations;
+using YouTubeV2.Api.Attributes;
+using YouTubeV2.Application.Model;
+using YouTubeV2.Application.Services;
+
+namespace YouTubeV2.Api.Controllers
+{
+    [ApiController]
+    [Route("donate")]
+    public class DonationController : IdentityControllerBase
+    {
+        private readonly IDonationService _donationService;
+
+        public DonationController(IDonationService donationService) 
+        {
+            _donationService = donationService;
+        }
+
+        [HttpPost("send")]
+        [Roles(Role.Simple, Role.Administrator)]
+        public async Task<ActionResult> SendDontaionAsync([FromQuery][Required] Guid id, [FromQuery][Required] decimal amount)
+        {
+            string? senderId = GetUserId();
+            if (senderId is null) return Forbid();
+
+            await _donationService.SendDonationAsync(senderId, id.ToString(), amount);
+
+            return Ok();
+        }
+
+        [HttpPost("withdraw")]
+        [Roles(Role.Creator)]
```

Comment:
are you sure this can only be a creator?

Reply:
what if creator changes their role to simple after getting some donations?

Reply:
according to spec (pdf) only simple can send money (to a creator), so I assume only creator can withdraw money (if simple "sends" moeny to our app he has to spend it there). If creator changes his role he can still change it back and get his money. But then anybody can change to creator and get their money back. So ok you convinced me I think